app/src/main/java/me/saket/dank/urlparser/RedditGalleryLink.kt

```diff
@@ -0,0 +1,82 @@
+package me.saket.dank.urlparser
+
+import android.os.Parcelable
+import com.squareup.moshi.JsonClass
+import kotlinx.android.parcel.Parcelize
+import me.saket.dank.utils.ImageWithMultipleVariants
+import net.dean.jraw.models.Submission
+import java.lang.IllegalStateException
+import java.lang.UnsupportedOperationException
+
+@JsonClass(generateAdapter = true)
+@Parcelize
+data class RedditGalleryLink(
+    val galleryUrl: String,
+    val imageList: List<RedditGalleryImageLink>
+): MediaAlbumLink<RedditGalleryImageLink>(), Parcelable {
+
+  companion object {
+    @JvmStatic fun extractImages(submission: Submission): Sequence<RedditGalleryImageLink> {
+      return submission.galleryData?.items?.asSequence()?.map {
+        val id = it.mediaId
+        val meta = submission.mediaMetadata?.get(id)
+
+        val (hqUrl, lqUrl, mediaType) = if (meta != null && meta.mime == "image/gif") {
+          val (url, mediaType) = meta.full.mp4Url
+              ?.let { u -> Pair(u, Type.SINGLE_VIDEO) }
+              ?: Pair("https://i.redd.it/$id.gif", Type.SINGLE_GIF)
+          Triple(url, null, mediaType)
+        } else {
+          val ext = when (meta?.mime) {
+            "image/jpeg", "image/jpg" -> "jpg"
+            "image/png" -> "png"
+            "image/webp" -> "webp"
+            else -> throw UnsupportedOperationException("Unknown mime type: ${meta?.mime}")
```

Comment:
Do we really want to cause a crash when we encounter unsupported mime type? I would suggest to throw exception only in debug builds and otherwise ignore such images or show errors.